server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/rewrite/task/RewriteCrossSpaceCompactionTask.java

```diff
@@ -172,18 +177,22 @@ private void executeCompaction()
   }
 
   private void updateTsFileResource() throws IOException {
+    seqTsFileResourceList.writeLock();
+    unseqTsFileResourceList.writeLock();
```

Comment:
when we change the resourceList, we need to get the writelock of TsFileManager, not the two list. 
The query only get the read lock of TsFileManager

Reply:
resolved